src/readFields.js

```diff
@@ -16,14 +16,20 @@ const readFields = (props, myFields, asyncValidate, isReactNative) => {
   const fieldObjects = {...myFields};
   fields.forEach(name => {
     const result = readField(form, name, undefined, fieldObjects, syncErrors, asyncValidate, isReactNative, props);
-    if (result.invalid) {
-      allValid = false;
-    }
-    if (result.dirty) {
-      allPristine = false;
-    }
-    if (result.error) {
-      errors = write(name, result.error, errors);
+    if (Array.isArray(result)) {
+      allValid = result.every(res => !res.invalid);
+      allPristine = result.every(res => !res.dirty);
+      errors = result.every(res => write(name, res.error, errors));
```

Comment:
Not quite. I added a few more tests, too, to confirm that everything is working.